roomserver/query/query.go

```diff
@@ -93,6 +93,10 @@ type RoomserverQueryAPIDatabase interface {
 	GetRoomVersionForRoom(
 		ctx context.Context, roomNID types.RoomNID,
 	) (gomatrixserverlib.RoomVersion, error)
+	// Look up the room NID that an event ID appears in.
+	GetRoomNIDForEventID(
+		ctx context.Context, eventID string,
+	) (types.RoomNID, error)
```

Comment:
Why are we doing this? In every single place we call this it's doing `event.EventID()`, why not just directly use `event.RoomID()` and use the existing `RoomNID(ctx context.Context, roomID string) (types.RoomNID, error)`? I don't want the API surface to get bigger than it already is!

Reply:
Have changed it. Some of this logic came from the old PR which admittedly doesn't make as much sense now. Thanks for pointing it out.